backend/src/main/java/me/quizzl/backend/services/QuizService.java

```diff
@@ -19,7 +20,35 @@ public Quiz addQuiz() {
         return quizRepository.save(quiz);
     }
 
+    public Quiz getQuizByID(UUID id) {
+        return quizRepository.getOne(id);
+    }
+
     public List<Quiz> getQuizzes() {
         return quizRepository.findAll();
     }
+    
+    public Grade gradeSubmission(Submission submission) {
```

Comment:
this is entirely subjective so feel free to just completely reject this if you don't wanna do it but I think submissions should get their own controller, repository, & service, and this method could be on that service instead

Reply:
This is the only one I didn't touch because I'm a bit busy but I do think this is a good idea. If I get more time I'll work on this and submit a new PR with the changes included. But if not then we can always throw it in a new user story in the future to work on because I do think it makes sense.

Reply:
Thanks for the suggestions btw I think they were good. The removal of the grade entity was already something I was considering but I wasn't sure if it would end up making things more or less complex in the future but I went ahead and removed it and I think it works better.